test/Reader/StandaloneExtensionManagerTest.php

```diff
@@ -80,4 +85,13 @@ public function testEachPluginRetrievalReturnsNewInstance($pluginName, $pluginCl
         $this->assertInstanceOf($pluginClass, $test);
         $this->assertNotSame($extension, $test);
     }
+
+    public function testPluginAddRemove()
```

Comment:
You need additional tests for expected exception in case of invalid extension class or object of valid extension.

Reply:
You need additional tests to ensure extensions derived from both allowed classes are accepted

Reply:
I've added tests for the exceptions, but the check in `add` is already checking for derived classes, because the class it checks for is abstract, so anything that is_a `Extension\AbstractEntry` is by definition derived from it, so passing a further derivation wouldn't be testing anything further.

Reply:
Those should be separate tests with descriptive names for what you are testing for
testAddAcceptsValidExtensionClasses()
testAddRejectsInvalidExtensions()
testAddRejectsInstanceOfValidExtension()

That way we have intent of the test and its implementation details rather than just asserting that current code runs without errors. It makes sure that if code changes at a later stage no intent will be lost.